akka-http-core/src/test/scala/akka/http/TestServer.scala

```diff
@@ -18,18 +20,35 @@ object TestServer extends App {
   implicit val system = ActorSystem("ServerTest", testConf)
   implicit val fm = ActorFlowMaterializer()
 
-  val binding = Http().bindAndHandleSync({
-    case HttpRequest(GET, Uri.Path("/"), _, _, _)      ⇒ index
-    case HttpRequest(GET, Uri.Path("/ping"), _, _, _)  ⇒ HttpResponse(entity = "PONG!")
-    case HttpRequest(GET, Uri.Path("/crash"), _, _, _) ⇒ sys.error("BOOM!")
-    case _: HttpRequest                                ⇒ HttpResponse(404, entity = "Unknown resource!")
-  }, interface = "localhost", port = 8080)
+  try {
+    val binding = Http().bindAndHandleSync({
+      case req @ HttpRequest(GET, Uri.Path("/"), _, _, _) if req.header[UpgradeToWebsocket].isDefined ⇒
```

Comment:
Yes, I wouldn't pollute it on this low level. We can hide this in the (still missing) routing API.